libs/datasets/weekly_hospitalizations.py

```diff
@@ -0,0 +1,55 @@
+import dataclasses
+import pandas as pd
+
+from libs.datasets import timeseries
+from datapublic.common_fields import CommonFields
+from libs.datasets.dataset_utils import AggregationLevel
+
+MultiRegionDataset = timeseries.MultiRegionDataset
+
+
+def _rolling_sum_7day(daily: pd.Series):
+    assert CommonFields.DATE in daily.index.names
+    return daily.rolling(7).sum()
+
+
+def calculate_weekly_column_from_daily(
+    dataset_in: MultiRegionDataset,
+    level_to_replace: AggregationLevel,
+    field_in: CommonFields,
+    field_out: CommonFields,
+) -> MultiRegionDataset:
+    """Calculate weekly occurrences from daily data using a rolling 7-day sum.
+
+    Args:
+        dataset_in: The dataset to update.
+        level_to_replace: Aggregation level to update data for.
+        field_in: Column to sum into weekly data, should have units of new daily occurrences (e.g. new daily hospital admissions).
+        field_out: Column to sum data into. Result will have units of new weekly occurrences (e.g. new weekly hospital admissions).  
+    """
+
+    state_ds = dataset_in.get_subset(aggregation_level=level_to_replace)
+    state_ts = state_ds.timeseries_bucketed.copy()  # copy to avoid SettingWithCopy warning.
```

Comment:
The `state_` prefix doesn't exactly make sense since `level_to_replace` could be a different level.  Perhaps `subset_ts` or `replace_ts` or something ?